app/app/src/main/java/com/example/myapplication/JoyStick.java

```diff
@@ -0,0 +1,341 @@
+import android.graphics.Paint;
+import android.graphics.RectF;
+import android.util.AttributeSet;
+import android.view.GestureDetector;
+import android.view.MotionEvent;
+import android.view.View;
+
+public class JoyStick extends View implements GestureDetector.OnGestureListener, GestureDetector.OnDoubleTapListener {
+
+
+    public static final int DIRECTION_CENTER = -1;
+    public static final int DIRECTION_LEFT = 0;
+    public static final int DIRECTION_LEFT_UP = 1;
+    public static final int DIRECTION_UP = 2;
+    public static final int DIRECTION_UP_RIGHT = 3;
+    public static final int DIRECTION_RIGHT = 4;
+    public static final int DIRECTION_RIGHT_DOWN = 5;
+    public static final int DIRECTION_DOWN = 6;
+    public static final int DIRECTION_DOWN_LEFT = 7;
+
+    public static final int TYPE_8_AXIS = 11;
+    public static final int TYPE_4_AXIS = 22;
+    public static final int TYPE_2_AXIS_LEFT_RIGHT = 33;
+    public static final int TYPE_2_AXIS_UP_DOWN = 44;
+
+    private JoyStickListener listener;
+    private Paint paint;
+    private RectF temp;
+    private GestureDetector gestureDetector;
+    private int direction = DIRECTION_CENTER;
+    private int type = TYPE_8_AXIS;
+    private float centerX;
+    private float centerY;
+    private float posX;
+    private float posY;
+    private float radius;
+    private float buttonRadius;
+    private double power = 0;
+    private double angle = 0;
+
+    //Background Color
+    private int padColor;
+
+    //Stick Color
+    private int buttonColor;
+
+    //Keeps joystick in last position
+    private boolean stayPut;
+
+    //Button Size percentage of the minimum(width, height)
+    private int percentage = 25;
+
+    //Background Bitmap
+    private Bitmap padBGBitmap = null;
+
+    //Button Bitmap
+    private Bitmap buttonBitmap = null;
+
+    public interface JoyStickListener {
+        void onMove(JoyStick joyStick, double angle, double power, int direction);
+
+        void onTap();
+
+        void onDoubleTap();
+    }
+
+    public JoyStick(Context context) {
+        this(context, null);
+    }
+
+    public JoyStick(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        paint = new Paint();
+        paint.setStyle(Paint.Style.FILL);
+        paint.setAntiAlias(true);
+        paint.setFilterBitmap(true);
+
+        temp = new RectF();
+
+        gestureDetector = new GestureDetector(context, this);
+        gestureDetector.setIsLongpressEnabled(false);
+        gestureDetector.setOnDoubleTapListener(this);
+
+        padColor = Color.WHITE;
+        buttonColor = Color.RED;
+
+        if (attrs != null) {
+            TypedArray typedArray = context.obtainStyledAttributes(attrs, R.styleable.JoyStick);
+            if (typedArray != null) {
+                padColor = typedArray.getColor(R.styleable.JoyStick_padColor, Color.WHITE);
+                buttonColor = typedArray.getColor(R.styleable.JoyStick_buttonColor, Color.RED);
```

Comment:
`Color.RED` should be replaced with `<color name="pink">#f66280</color>` that is found in the `colors.xml` file in `values`. `Color.RED` does not align with the rest of the UI design.